src/MoBi.Presentation/Tasks/Edit/EditTaskForContainer.cs

```diff
@@ -86,6 +94,66 @@ public override void Save(IContainer container)
          exportContainer(container, fileName);
       }
 
+      private void addIndividualParametersToContainers(IndividualBuildingBlock individual, IContainer container)
+      {
+         if (individual == null || !individual.Any())
+            return;
+
+         var allSubContainers = container.GetAllContainersAndSelf<IContainer>();
+         allSubContainers.Each(subContainer =>
+         {
+            var subContainerPath = container.ParentPath.Concat(_objectPathFactory.CreateAbsoluteObjectPath(subContainer));
+            addIndividualParametersToContainer(individual, subContainer, new ObjectPath(subContainerPath));
+         });
+      }
+
+      private void addIndividualParametersToContainer(IndividualBuildingBlock individual, IContainer container, ObjectPath containerPath)
+      {
+         var individualParametersToExport = individual.Where(individualParameter => individualParameter.ContainerPath.StartsWith(containerPath)).ToList();
+
+         // Ensure that the distributed parameters are added to the container first. After all distributed parameters are added, then the other parameters can be added.
+         individualParametersToExport.Where(x => x.DistributionType != null).Each(x => addIndividualParameterToContainer(x, container, containerPath));
+         individualParametersToExport.Where(x => x.DistributionType == null).Each(x => addIndividualParameterToContainer(x, container, containerPath));
+      }
+
+      private void addIndividualParameterToContainer(IndividualParameter individualParameter, IContainer container, string containerPath)
+      {
+         var targetContainer = findTargetContainer(individualParameter.ContainerPath, container, containerPath);
+         // The target container for this parameter is not found - skip
+         if (targetContainer == null)
+            return;
+
+         var parameterToAdd = createParameter(individualParameter);
+         var existingParameter = targetContainer.FindByName(parameterToAdd.Name);
+
+         if (existingParameter != null)
+            targetContainer.RemoveChild(existingParameter);
+
+         targetContainer.Add(parameterToAdd);
+      }
+
+      private IParameter createParameter(IndividualParameter individualParameter)
+      {
+         var parameterToAdd = _individualParameterToParameterMapper.MapFrom(individualParameter);
+
+         if (individualParameter.Formula != null)
+            parameterToAdd.Formula = individualParameter.Formula;
+         if (individualParameter.Value.HasValue)
+            parameterToAdd.Formula = _formulaFactory.ConstantFormula(individualParameter.Value.Value, individualParameter.Dimension);
```

Comment:
Questions on this:

a) shouldn't it be encapsulated in `IndividualParameterToParameterMapper.MapFrom()` ?
b) don't we need to **clone** the formula in `parameterToAdd.Formula = individualParameter.Formula;` ?
c) the second assignment overrides the first one (if the condition is true), so I would rather do it as
```
if (individualParameter.Value.HasValue)
...
else if (individualParameter.Formula != null)
...
```

Reply:
Well, now I'm not so sure this should be encapsulated. We are using the same mapper for model creation, so the cloning works differently. I'll put the else-if here and add a comment why this is not encapsulated.

Reply:
ok